src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -229,7 +228,15 @@ private UnboundLambda BindAnonymousFunction(CSharpSyntaxNode syntax, DiagnosticB
                 }
             }
 
-            var lambda = new UnboundLambda(syntax, this, refKinds, types, names, isAsync);
+            ModifierUtils.CheckModifiers(
+                ModifierUtils.ToDeclarationModifiers(syntax.Modifiers, diagnostics),
+                allowedModifiers: DeclarationModifiers.Static | DeclarationModifiers.Async,
+                GetLocationForDiagnostics(syntax),
+                diagnostics,
+                syntax.Modifiers,
+                out _);
```

Comment:
Does this have any effect? (It looks like parsing ignores any modifiers other than `static` and `async`?

Reply:
Yes.  it ensures that if there are *duplicate* modifiers (like `static static delegate () { }`) that we report the issue.

Reply:
It looks like `ModifierUtils.ToDeclarationModifiers()` reports duplicates. Consider removing the call to `CheckModifiers`.

---
In reply to: [335695721](https://github.com/dotnet/roslyn/pull/39121#discussion_r335695721) [](ancestors = 335695721)

Reply:
Good point.  Thanks!